src/org/labkey/test/components/ui/entities/ParentEntityEditPanel.java

```diff
@@ -169,76 +156,39 @@ public boolean isAddButtonEnabled()
      */
     protected ReactSelect getAddNewTypeCombo()
     {
-        int numOfTypes = numberOfTypeFields();
-
-        ReactSelect typeCombo = getTypeCombo(_parentType.getType() + " Type " + numOfTypes);
+        var numOfTypes = getAllTypeCombo().size();
+        var typeCombo = getTypeCombo(numOfTypes);
 
         // If the 'last' combo in the list contains the text "Select a..." it can be used to add a new type.
         // If it does not contain that then the addButton must be clicked.
-        if(!typeCombo.getSelections().contains("Select a Source Type ...") &&
-                        !typeCombo.getSelections().contains("Select a Parent Type ..."))
+        if (typeCombo.hasSelection())
         {
             // Since there is already a parent need to click the "Add" button to add a new one.
             elementCache().addButton.click();
 
             // Now need to go find the new combo but wait until it shows up.
-            WebDriverWrapper.waitFor(()-> numberOfTypeFields() > numOfTypes,
+            WebDriverWrapper.waitFor(()-> getAllTypeCombo().size() > numOfTypes,
                     "The new type field did not show up in a timely fashion.",
                     1_000);
 
-            typeCombo = ReactSelect.finder(getDriver())
-                    .followingLabelWithSpan(_parentType.getType() + " Type " + (numOfTypes + 1))
-                    .find(this);
-
+            typeCombo = getTypeCombo(numOfTypes + 1);
         }
 
         return typeCombo;
-
-    }
-
-    /**
-     * Get the type combos by it's index (zero based). This would include any 'add new' combos.
-     *
-     * @return A combo at the given position in the collection of combos.
-     */
-    protected ReactSelect getTypeCombo(int index)
-    {
-        return getAllTypeCombo().get(index);
     }
 
     /**
      * Get the type combos by its label. This is the label to the left of the control.
      *
      * @return A combo beside a label with the given text.
      */
-    protected ReactSelect getTypeCombo(String labelText)
```

Comment:
If there's a way to preserve the ability to specify which one by label text, that would be preferable to using index (which is more fragile and obtuse) if that's the tradeoff being made.  If we're using index, usually it's because other (more deterministic) means of identifying the target weren't clear or available